docs/source/api/apollo-server.mdx

```diff
@@ -72,7 +74,7 @@ For an example, see [Define your GraphQL schema](../getting-started/#step-3-defi
 
 A map of functions that populate data for individual schema fields. Can also be an array of multiple maps that are merged.
 
-**Required** unless you provide [`schema`](#schema).
+**Required** unless you provide a [`schema`](#schema) or a [`gateway`](#gateway).
```

Comment:
AS will actually work with `typeDefs` only and no resolvers - they will just be the default resolvers (i.e. whatever the name of the field is must have a matching property on the incoming parent value).

Just about any _useful_ schema will have resolvers implemented, though, so it's not terrible to leave as is. Technically, this is incorrect though.

```ts
interface ApolloServerOptionsWithTypeDefs<TContext extends BaseContext>
  extends ApolloServerOptionsBase<TContext> {
  // These two options are always only passed directly through to
  // makeExecutableSchema. (If you don't want to use makeExecutableSchema, pass
  // `schema` instead.)
  typeDefs: IExecutableSchemaDefinition<TContext>['typeDefs'];
  resolvers?: IExecutableSchemaDefinition<TContext>['resolvers'];
  gateway?: undefined;
  schema?: undefined;
}
```

Reply:
That is an excellent point! I think we can split the difference here with a sentence like:

> While technically optional, if you are providing `typeDefs`, it is strongly recommended that you provide `resolvers` as well.

Reply:
I like it!